frontend/src/lib/services/User-Listing/service.ts

```diff
@@ -0,0 +1,28 @@
+import api, { EP } from "../../../../network/api";
+import { UserListingType, UserListingInput } from "./types";
+
+type Response<T> = Promise<{ data: T; error: string }>;
+
+const UserListingService = () => ({
+  create: (
+    userListing: UserListingInput
+  ): Response<{ listing: UserListingType }> =>
+    api.post(EP.userlistings.createListing, userListing),
+
+  getById: (userlistingId: string): Response<{ listing: UserListingType }> =>
+    api.get(EP.userlistings.getUserListingById(userlistingId)),
+
+  getAll: (): Response<{ userlistings: UserListingType[] }> =>
+    api.get(EP.userlistings.getAllUserListings),
+
+  update: (
+    userlistingId: string,
+    userListing: Partial<UserListingInput>
```

Comment:
nonblocking: why is this returning a "Partial" UserListingInput type and not just  a UserListingInput type